Zend/zend_compile.c

```diff
@@ -3678,25 +3678,39 @@ void zend_compile_call_common(znode *result, zend_ast *args_ast, zend_function *
 {
 	zend_op *opline;
 	uint32_t opnum_init = get_next_op_number() - 1;
-	uint32_t arg_count;
-	bool may_have_extra_named_args;
 
-	arg_count = zend_compile_args(args_ast, fbc, &may_have_extra_named_args);
+	if (args_ast->kind != ZEND_AST_CALLABLE_CONVERT) {
+		bool may_have_extra_named_args;
+		uint32_t arg_count = zend_compile_args(args_ast, fbc, &may_have_extra_named_args);
 
-	zend_do_extended_fcall_begin();
+		zend_do_extended_fcall_begin();
 
-	opline = &CG(active_op_array)->opcodes[opnum_init];
-	opline->extended_value = arg_count;
+		opline = &CG(active_op_array)->opcodes[opnum_init];
+		opline->extended_value = arg_count;
 
-	if (opline->opcode == ZEND_INIT_FCALL) {
-		opline->op1.num = zend_vm_calc_used_stack(arg_count, fbc);
-	}
+		if (opline->opcode == ZEND_INIT_FCALL) {
+			opline->op1.num = zend_vm_calc_used_stack(arg_count, fbc);
+		}
+
+		opline = zend_emit_op(result, zend_get_call_op(opline, fbc), NULL, NULL);
+		if (may_have_extra_named_args) {
+			opline->extended_value = ZEND_FCALL_MAY_HAVE_EXTRA_NAMED_PARAMS;
+		}
+		zend_do_extended_fcall_end();
+	} else {
+		opline = &CG(active_op_array)->opcodes[opnum_init];
+		opline->extended_value = 0;
+
+		if (opline->opcode == ZEND_NEW) {
+		    zend_error_noreturn(E_COMPILE_ERROR, "cannot create Closure from call to constructor");
```

Comment:
```suggestion
		    zend_error_noreturn(E_COMPILE_ERROR, "Cannot create Closure for new expression");
```
This is missing a test